src/pydash/objects.py

```diff
@@ -2422,7 +2422,7 @@ def update_with(obj, path, updater, customizer=None):  # noqa: C901
         if isinstance(token, PathToken):
             key = token.key
             default_factory = pyd.get(tokens, [idx + 1, "default_factory"], default=default_type)
-        else:
+        else:  # pragma: no cover
```

Comment:
If the `else` case is no longer valid and `isinstance(token, PathToken)` is always true, then I think we could remove the if-statement and just have whatever is under the main `if` section.

Unless there was a reason to leave them in?

Reply:
Yeah makes sense. I removed them.

I left them just to be safe, but I guess if it happens it would probably be a rare edge case. The test cases cover a lot of cases and every passes so it should be safe enough.

Reply:
> would probably be a rare edge case

From looking at the code, the `tokens` are from `to_path_tokens()` which always returns a `PathToken` list, right? So should never be any other type. Unless I missed something in `to_path_tokens()` where a different type could be included?

Reply:
`to_path_tokens` can return something else than a `PathToken` if it is a single element like `"0"` of `"some_key"`.
But I guess we should probably make `to_path_tokens` always return a list of `PathToken` all the time. It would be more consistant and easier to reason about.
I added a commit with it. Let me know what you think